### PR TITLE
Fix group_bug_information and add a retry mechanism if we go past the allowable transaction limit

### PR DESCRIPTION
**Pull request by rmistry**
* group_bug_information is an int and not a str like bug_information
* If we go past the allowable transaction limit then retry the put with half the specified batch size

### CODE REVIEW

**Comment by rmistry** [comment]
PTAL @jonathanmetzman @letitz @oliverchang 
CC @micahbales

File: src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py [hunk 1]
```diff
@@ -66,35 +70,53 @@ def execute(args):
...
-
-      count_of_updated += len(testcases)
-      print(f'Updated {len(testcases)}. Total {count_of_updated}')
-      testcases = []
-
-  if args.non_dry_run and len(testcases) > 0:
-    ndb.put_multi(testcases)
-    count_of_updated += len(testcases)
-    print(f'Updated {len(testcases)}. Total {count_of_updated}')
+    else:
```
Comment (letitz) [inline src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py:118]: What does this `else` clause do?

File: src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py [hunk 2]
```diff
@@ -66,35 +70,53 @@ def execute(args):
...
+  try:
+    ndb.put_multi(testcases)
+  except Exception as e:
+    if PAYLOAD_SIZE_ERROR in str(e) and len(testcases) > 1:
+      half_batch_size = len(testcases)//2
+      print('Reached payload size limit. Retrying batch put with half the '
+            f'specified batch size: {half_batch_size}')
       try:
-        ndb.put_multi(testcases)
+        ndb.put_multi(testcases[0:half_batch_size])
```
Comment (letitz) [inline src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py:108]: Tiny nit

```suggestion
        ndb.put_multi(testcases[:half_batch_size])
```

**Review by letitz — COMMENTED** [review]
One question and a tiny nit.

File: src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py (hunk 1 above)
Comment (rmistry) [reply src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py:118]: Reraises the error if `if PAYLOAD_SIZE_ERROR in str(e) and len(testcases) > 1:` is not satisfied.

File: src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py (hunk 2 above)
Comment (rmistry) [reply src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py]: Done.

File: src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py [hunk 3]
```diff
@@ -66,35 +70,53 @@ def execute(args):
...
+        ndb.put_multi(testcases[half_batch_size:])
       except Exception as e:
-        if '400 Request payload size exceeds the limit' in str(e):
+        if PAYLOAD_SIZE_ERROR in str(e):
           print(f'Got exception: {e}')
           print('Opening debugger to investigate further:')
           # pylint: disable=forgotten-debug-statement
           import pdb
           pdb.set_trace()
         raise
```
[the remark this answers is not in the document]
Comment (rmistry) [reply src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py]: Removed.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Pull request merged**